crates/accelerate/src/unitary_synthesis.rs

```diff
@@ -354,26 +379,33 @@ fn py_run_main_loop(
                     ref_qubits,
                     &coupling_edges,
                     target,
+                    basis_gates.clone(),
                     approximation_degree,
                     natural_direction,
+                    pulse_optimize,
                     &mut out_dag,
                     out_qargs,
                     apply_original_op,
                 )?;
             }
             // Run 3q+ synthesis
             _ => {
-                let qs_decomposition: &Bound<'_, PyAny> = imports::QS_DECOMPOSITION.get_bound(py);
-                let synth_circ = qs_decomposition.call1((unitary.into_pyarray(py),))?;
-                let synth_dag = circuit_to_dag(
-                    py,
-                    QuantumCircuitData::extract_bound(&synth_circ)?,
-                    false,
-                    None,
-                    None,
-                )?;
-                let out_qargs = dag.get_qargs(packed_instr.qubits);
-                apply_synth_dag(py, &mut out_dag, out_qargs, &synth_dag)?;
+                if basis_gates.is_empty() && matches!(target, None) {
```

Comment:
Minor question: is there a benefit of writing `matched!(target, None)` over `target.is_none()`?

Reply:
I don't think so, this was me just being rusty in Rust when I wrote it :) I actually prefer `target.is_none()`